src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/pointer"
+)
+
+func TestValidateUpdate(t *testing.T) {
+	var replicas1 int32 = 1
+	var replicas2 int32 = 2
+
+	redpandaCluster := &v1alpha1.Cluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:		"RedpandaCluster",
+			APIVersion:	"core.vectorized.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:		"test",
+			Namespace:	"",
+		},
+		Spec: v1alpha1.ClusterSpec{
+			Replicas:	pointer.Int32Ptr(replicas2),
+			Configuration: v1alpha1.RedpandaConfig{
+				KafkaAPI: v1alpha1.SocketAddress{Port: 123},
+			},
+		},
+	}
+
+	updatedCluster := redpandaCluster.DeepCopy()
+	updatedCluster.Spec.Replicas = &replicas1
+	updatedCluster.Spec.Configuration = v1alpha1.RedpandaConfig{
+		KafkaAPI: v1alpha1.SocketAddress{Port: 1234},
+	}
+
+	err := updatedCluster.ValidateUpdate(redpandaCluster)
+	if err == nil {
+		t.Fatalf("expecting validation error but got none")
+	}
+
+	// verify the error causes contain all expected fields
+	statusError := err.(*apierrors.StatusError)
+	expectedFields := []string{field.NewPath("spec").Child("configuration").String(), field.NewPath("spec").Child("replicas").String()}
+	for _, ef := range expectedFields {
+		found := false
+		for _, c := range statusError.Status().Details.Causes {
+			if ef == c.Field {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("expecting failure on field %s but have %v", ef, statusError.Status().Details.Causes)
+		}
+	}
+}
+
+func TestValidateUpdate_NoError(t *testing.T) {
+	var replicas2 int32 = 2
+
+	redpandaCluster := &v1alpha1.Cluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:		"RedpandaCluster",
+			APIVersion:	"core.vectorized.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:		"test",
+			Namespace:	"",
+		},
+		Spec: v1alpha1.ClusterSpec{
+			Replicas:	pointer.Int32Ptr(replicas2),
+			Configuration: v1alpha1.RedpandaConfig{
+				KafkaAPI: v1alpha1.SocketAddress{Port: 123},
+			},
+		},
+	}
+
+	err := redpandaCluster.ValidateUpdate(redpandaCluster)
+	assert.NoError(t, err, "same object")
+
+	// Scale up
```

Comment:
NIT: it could be wrapped in subtest.

Reply:
subtest you mean different test? 🤔 I thought there's very little shared I just did it like this but I am open to refactoring

Reply:
The answer is in https://github.com/vectorizedio/redpanda/pull/665/files#r581164986